### ISSUE TITLE
Annoying 30s bootup pause on EC2 with HVM "Timeout connecting to device"

### ISSUE THREAD

**Issue by pwaller** [coreos/bugs#208]
There is annoying pause before "xenbus_probe_frontend: Timeout connecting to device: device/vfb/0".

Appears to be related to [Xen bug 43](http://bugs.xenproject.org/xen/bug/43). Here is some [discussion on the debian list](https://lists.debian.org/debian-cloud/2014/11/msg00002.html) from Nov 3rd.

Here's an excerpt from the console:

```
[    0.926813] NET: Registered protocol family 10
[    0.929792] NET: Registered protocol family 17
[    0.932351] Loading compiled-in X.509 certificates
[    0.935626] Loaded X.509 cert 'Magrathea: Glacier signing key: e8e64520335bdabd7d6234e481ed82c777107866'
[    0.940717] registered taskstats version 1
[    1.780113] tsc: Refined TSC clocksource calibration: 2499.999 MHz
[    6.043103] xenbus_probe_frontend: Waiting for devices to initialise: 25s...20s...15s...10s...5s...0s...
[   30.948100] 
[   30.949564] xenbus_probe_frontend: Timeout connecting to device: device/vfb/0 (local state 3, remote state 1)
[   30.954568] xenbus_probe_frontend: Device with no driver: device/vbd/51712
[   30.957841] xenbus_probe_frontend: Device with no driver: device/vif/0
[   30.961155] rtc_cmos 00:02: setting system clock to 2014-12-12 16:56:21 UTC (1418403381)
[   30.972348] Freeing unused kernel memory: 21236K (ffffffff81ae6000 - ffffffff82fa3000)
[   30.978569] Write protecting the kernel read-only data: 10240k
[   30.985240] Freeing unused kernel memory: 1060K (ffff8800014f7000 - ffff880001600000)
[   30.994618] Freeing unused kernel memory: 1836K (ffff880001835000 - ffff880001a00000)
[   31.001640] random: systemd urandom read with 79 bits of entropy available
[   31.005571] systemd[1]: systemd 215 running in system mode. (-PAM -AUDIT -SELINUX +IMA -SYSVINIT +LIBCRYPTSETUP -GCRYPT -ACL -XZ +SECCOMP -APPARMOR)

```

**Comment by jedsmith** [comment]
I'm hitting this too. Depending on where the Xen bug is, this might be unsolvable in CoreOS; however, it sounds like a driver issue on the client side, which might be fixable. I'm investigating.

**Comment by jedsmith** [comment]
Demoting Xen's FBDEV to a module alleviates the issue. Issued a PR upstream.

**Comment by pwaller** [comment]
The release notes say that this is fixed in the alpha.

I just tried ami-73e98b04 and my dmesg log still shows the pause. Please reopen.

```
CoreOS alpha (660.0.0)
pwaller@ip-10-1-29-128 ~ $ dmesg
<snip>
[    0.000000] Linux version 4.0.0 (buildbot@ip-10-204-3-57) (gcc version 4.8.3 (Gentoo Hardened 4.8.3 p1.1, pie-0.5.9) )
<snip>
[    6.844083] xenbus_probe_frontend: Waiting for devices to initialise: 25s...20s...15s...10s...5s...0s...

[   31.851233] xenbus_probe_frontend: Timeout connecting to device: device/vfb/0 (local state 3, remote state 1)
<snip>
```

**Comment by pwaller** [comment]
![screenshot from 2015-04-24 12 11 28](https://cloud.githubusercontent.com/assets/438648/7317693/57c3168c-ea7b-11e4-8dad-857bec1c4a73.png)

**Comment by jedsmith** [comment]
Yep, looks like the module gets loaded now, which didn't happen in my testing.

**Comment by jedsmith** [comment]
@pwaller Try banning `xen_fbfront` on your kernel's command line in grub:

    modprobe.blacklist=xen_fbfront

I just wrote that as a potential fix in jedsmith/coreos-overlay@30c493e, but I haven't tested yet.

**Comment by pwaller** [comment]
@jedsmith apologies for my ignorance, but I can't figure out how to do that quickly on EC2. Usually I'd look for grub configuration in /boot/grub, but /boot is empty. Any hints? I'd love to see a pull request to add this fix! :)

**Comment by pwaller** [comment]
Can confirm fixed in 681.0.0, thanks @jedsmith !

**Comment by jedsmith** [comment]
Thanks for the confirmation. Glad to hear it!
